### PR TITLE
Remove use of grunt-mocha

### PR DESCRIPTION
**Pull request by benhutchins**
Removed the use of the module `grunt-mocha`, wrapping it's features directly into `grunt-js-test`.

grunt-mocha is not actively maintained. It's using grunt-lib-phantomjs 0.4, they're on 0.7. grunt-lib-phantomjs is what provides the phantomjs library, this results in the tests running in an old version of phantomjs and not getting all the latest and greatest features.

As part of this minor rewrite, I've also removed the two useless tasks (`js-test-server-keepalive` and `js-test-save`). `js-test-server-keepalive` is now built into `js-test-server`, and `js-test-save` is supposed to save the results of a coverage report after grunt-mocha had finished. Since the grunt-mocha task is now wrapped into the `js-test` task, the `js-test-save` task could also be wrapped into it. The new `tasks/lib/save.js` file handles saving the coverage reports.

Other misc cleanups include:
 * Reduces repeated code to handle the `options` object for `js-test` and `js-test-server`
 * Fixed issue if you were generating a coverage report, after all tests finished you did not get a final stats print out (see changes to `views/deps/shared.js`)
 * Removed use of included mocha library, made mocha an npm module and that is now included.
 * Removed npm module `phantomjs` which was actually unused, and was just being wastefully downloaded.

This is all in preparation for adding Selenium support to `grunt-js-test` using `selenium-webdriver` or `webdriverio`. Going to make it so you can decide whether to run your tests in phantomjs or configure a selenium server and run it in browsers such as Chrome, Firefox, IE, Safari, Opera, etc. I'd also like to start the discussion of the better library. It'll be easier to implement support using `selenium-webdriver`, the official selenium library for Node.js - however, webdriverio supports the full WebDriver Wire Protocol allowing it to also work with https://www.browserstack.com/ and http://testingbot.com/. Once this is completed, grunt-js-test can via a CI and easily run all the tests in multiple browsers, then generate reports.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -40,13 +40,13 @@
...
     "body-parser": "^1.6.5",
     "express": "4.x",
-    "grunt-mocha": "0.4.x",
+    "grunt-lib-phantomjs": "^0.7.0",
     "istanbul": "^0.3.0",
     "jade": "1.x",
     "lodash": "^2.4.1",
-    "moment": "^2.8.1",
     "minimatch": "^2.0.4",
-    "phantomjs": "1.9.13",
```
Comment (texclayton) [inline package.json]: Don't remove this line.  Newer versions of phantomjs fail to install nearly 25% of the time in automated Jenkins jobs.  The workaround is to force the use of this version of phantomjs (there's a shrinkwrap.json for this reason also).

**Comment by texclayton** [comment]
This sounds good.  I was thinking about Selenium support last week.  Also, Karma.
I'll try to have this merged by Friday evening.

File: package.json (hunk 1 above)
Comment (benhutchins) [inline package.json]: Ah, I didn't see any notes about that, and nothing was using that copy, or so it seemed. I'll revise my commit tomorrow.

**Comment by benhutchins** [comment]
Karma is similar to grunt-js-test in some ways. Not sure they'd work together well, what are your thoughts there?

**Comment by texclayton** [comment]
I was wondering how they might work together.  I haven't had a chance to look at Karma in depth, but like you said, they are similar.  Selenium support looks like a better ROI right now.
Start an Issue about Selenium, please.

**Comment by texclayton** [comment]
I believe this fixes Issue #3.  Probably also partially addresses Issue #2

**Comment by benhutchins** [comment]
It does fix #3 , will be using the latest release of Mocha and also allows updating Mocha in the future to be easier, you can use `npm outdated` to look for updates in the future.

As for #2, this is not really solved. If Mocha is changed to a shared dependency it would be, but it'll require a rewrite of the `server.js` to use `require.resolve` to get the path of mocha since it may not be apart of grunt-js-test directly. This probably makes sense to do, simply hadn't thought of it.

File: package.json (hunk 1 above)
Comment (texclayton) [inline package.json]: Ug.  I just realized that this PR includes a newer version of grunt-lib-phantomjs, which pulls in phantomjs-1.9.17, rendering this line in package.json useless.

Comment (benhutchins) [inline package.json]: Do we have to go back to an older version of grunt-lib-phantomjs?

Comment (texclayton) [inline package.json]: No, let's leave it.  I can't update the npm package yet, though.

**Pull request merged**